muni_portal/core/signals.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from django_q.tasks import async_task
+from pywebpush import webpush, WebPushException
+
+from muni_portal.core.models import WebPushNotification, WebPushSubscription, WebPushNotificationResult
+
+logger = logging.getLogger(__name__)
+
+
+def queue_send_webpush_notification(notification_id):
+    notification = WebPushNotification.objects.get(id=notification_id)
+    for subscription in WebPushSubscription.objects.all():
+        result = WebPushNotificationResult(subscription=subscription, notification=notification)
+        try:
+            response = webpush(
+                subscription.serialize(), notification.serialize(),
+                vapid_private_key=settings.VAPID_PRIVATE_KEY,
+                vapid_claims={"sub": f"mailto:{settings.DEFAULT_FROM_EMAIL}"}
+            )
+            result.status_code = response.status_code
+        except WebPushException as e:
+            logger.error(f"Can't send web push notification to subscription #{subscription.id}, error {e}")
+            result.message = e.message
+            result.status_code = 500
```

Comment:
I don't think we should fake status codes as 500 - that could be confusing - this should rather be null if there's no status code.